Program.cs

```diff
@@ -19,13 +19,53 @@ public static List<string> ReadWordsFromFile(string filePath)
         return words;
     }
 
-    public static void Main()
+    public static List<string> RemoveDuplicateWords(List<string> words)
     {
-        List<string> wordsList = ReadWordsFromFile("C:\\Users\\Zoran\\Desktop\\Dario_Internship\\#3_Words\\text.txt");
+        HashSet<string> uniqueWords = new HashSet<string>(words);
+        return new List<string>(uniqueWords);
+    }
 
-        foreach (string word in wordsList)
+    public static Dictionary<string, int> CountWordFrequency(List<string> words)
+    {
+        Dictionary<string, int> wordFrequency = new Dictionary<string, int>(); 
+        foreach (string word in words)
         {
-            Console.WriteLine(word);
+            if (wordFrequency.ContainsKey(word)) wordFrequency[word]++;
+            else wordFrequency[word] = 1;
         }
+        return wordFrequency;
+    }
+
+    public static void PrintSummary(List<string> wordsList, Dictionary<string, int> wordFrequency)
+    {
+        int totalWords = wordsList.Count;
+        int uniqueWords = wordFrequency.Count;
+
+        Console.WriteLine($"Total number of words in the document: {totalWords}");
+        Console.WriteLine($"Number of unique words: {uniqueWords}");
+
+        int maxFrequency = wordFrequency.Values.Max();
+        List<string> mostFrequentWords = wordFrequency.Where(pair => pair.Value == maxFrequency)
+                                                     .Select(pair => pair.Key)
+                                                     .ToList();
+        string mostFrequentWordsStr = string.Join(", ", mostFrequentWords);
```

Comment:
This is working fine, but you don't really see the `.Join` method all that often (no need to change anything).
We could use a few different approaches to do this, most notably _string interpolation_ and concatenation using the `+` operator. For the most part it is a personal preference thing but I'd dare to say that string interpolation is the most common way to handle this sort of thing. See the [reference](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated).

Reply:
Got it